src/zenml/integrations/github/orchestrators/github_actions_orchestrator.py

```diff
@@ -0,0 +1,450 @@
+                    "The GitHub Actions orchestrator requires a remote "
+                    f"container registry, but the '{container_registry.name}' "
+                    "container registry of your active stack points to a local "
+                    f"URI '{container_registry.uri}'. Please make sure stacks "
+                    "with a GitHub Actions orchestrator always contain remote "
+                    "container registries."
+                )
+
+            # for component in stack.components.values():
+            #     if component.local_path:
+            #         return False, (
+            #             "The GitHub Actions orchestrator runs pipelines on "
+            #             "remote GitHub Actions runners, but the "
+            #             f"'{component.name}' {component.TYPE.value} of your "
+            #             "active stack is a local component. Please make sure "
+            #             "to only use remote stack components in combination "
+            #             "with the GitHub Actions orchestrator. "
+            #         )
+
+            return True, ""
+
+        return StackValidator(
+            required_components={StackComponentType.CONTAINER_REGISTRY},
+            custom_validation_function=_validate_local_requirements,
+        )
+
+    def get_docker_image_name(self, pipeline_name: str) -> str:
+        """Returns the full docker image name including registry and tag."""
+        container_registry = Repository().active_stack.container_registry
+        assert container_registry  # should never happen due to validation
+        return f"{container_registry.uri}/zenml-github-actions:{pipeline_name}"
+
+    def _docker_login_step(
+        self,
+        container_registry: BaseContainerRegistry,
+    ) -> Optional[Dict[str, Any]]:
+        """GitHub Actions step for authenticating with the container registry.
+
+        Args:
+            container_registry: The container registry which (potentially)
+                requires a step to authenticate.
+
+        Returns:
+            Dictionary specifying the GitHub Actions step for authenticating
+            with the container registry if that is required, `None` otherwise.
+        """
+        if (
+            isinstance(container_registry, GitHubContainerRegistry)
+            and container_registry.automatic_token_authentication
+        ):
+            # Use GitHub Actions specific placeholder if the container registry
+            # specifies automatic token authentication
+            username = "{{ github.actor }}"
+            password = "{{ secrets.GITHUB_TOKEN }}"
+        elif container_registry.requires_authentication:
+            username = cast(str, container_registry.username)
+            password = cast(str, container_registry.password)
+        else:
+            return None
+
+        return {
+            "name": "Authenticate with the container registry",
+            "uses": "docker/login-action@v1",
```

Comment:
This sort of thing should be parameterized I think. Maybe a constant at the top?

Reply:
Just a constant at least?